apps/src/lab2/views/Lab2ShareDialogWrapper.tsx

```diff
@@ -81,11 +96,22 @@ const Lab2ShareDialogWrapper: React.FunctionComponent<
 
   const onGoBack = () => {
     setDialogPanel('share');
-    // If the project was submitted successfully, the submission status is updated.
-    if (channelId && projectType) {
-      getSubmissionStatus(channelId, projectType).then(response =>
-        setSubmissionStatus(response)
-      );
+    // If the project was submitted successfully, the submission status is updated (only for signed-in users).
```

Comment:
Would it make sense for the `submit` API to return the newly updated status so we don't have to fetch it again when returning to the previous dialog?

Reply:
My initial feeling is that I like having these actions distinct and separate.

Reply:
Makes sense; in that case can we extract this fetching and error handling code since it's the same here and in the above `useEffect`?

Reply:
Updated!